### PR TITLE
feat(tui): @ file-mention picker in the cockpit composer

### LINKED ISSUE #1699: TUI cockpit: @ file mention picker in the composer
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Split out from the cockpit-in-TUI follow-ups epic #1113.

## Problem

The TUI cockpit composer has no `@` file mention picker. The web composer lets you type `@` and pick a project file from a fuzzy-filtered list backed by the daemon. The TUI cannot do this, so referencing a file means typing the full path by hand.

## What's there today

The daemon endpoint already exists:

- `src/server/api/cockpit.rs:584` (`cockpit_files`) walks the project tree and returns `FilesResponse { files, truncated }`, capped at 5000 entries, served at `GET /api/sessions/{id}/cockpit/files`.

The TUI already holds an HTTP client to call it:

- `src/tui/cockpit_view/state.rs:13` (`CockpitViewState`) owns an `HttpClient` / `DaemonEndpoint`.

The composer has no `@` handling and no file cache:

- `src/tui/cockpit_view/input.rs:71` routes characters to `Intent::Compose` with no `@` interception.
- `src/tui/cockpit_view/reducer.rs` stores no file list.

Web reference to mirror:

- `web/src/components/cockpit/Composer.tsx:249` builds `fileAdapter` from a files index and applies a fuzzy filter.
- `web/src/components/cockpit/Composer.tsx:468` renders the `@` picker via `TriggerPopover`, inserting the selected path.
- `web/src/components/cockpit/Composer.tsx:507` placeholder reads "Type @ for files, / for commands".

## Proposal

On `@` in the composer, fetch `cockpit_files` through the existing `HttpClient`, cache the result in view state, fuzzy-filter as the user types, render a picker in `render.rs`, and insert the selected path on Enter via `input.rs`.

Touch points: `src/tui/cockpit_view/state.rs`, `src/tui/cockpit_view/input.rs`, `src/tui/cockpit_view/render.rs`.

## User stories / regression tests

Implementation MUST add tests covering each story below; PR is not done without them.

- As a cockpit TUI user, when I type `@`, then a picker lists project files fetched from the daemon and narrows as I keep typing.
  - Suggested test layer: `tests/e2e/` against a live daemon (the files endpoint must answer).
- As a cockpit TUI user, when I select a file, then its path is inserted at the cursor in the composer.
  - Suggested test layer: `src/tui/cockpit_view` unit test on insertion behavior.

## Open questions

- Cache policy: fetch once per session, or refetch on each `@`, or TTL?
- Should fuzzy ranking match the web's adapter exactly, or is "good enough" fine?
- Surface `truncated` (over 5000 files) to the user, or silently cap?

## Out of scope

- Slash-command picker (separate follow-up).
- Image or attachment mentions.

## Cross-references

- Parent epic: #1113.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

The TUI cockpit composer had no `@` file-mention picker, so referencing a file meant typing the full path by hand. The web composer already does this, backed by the daemon's `cockpit/files` index. This brings the same affordance to the TUI.

Typing `@` in the composer now opens a picker listing the session's workspace files, fetched once per session from `GET /api/sessions/{id}/cockpit/files` and cached for the session lifetime. Keep typing to fuzzy-filter (prefix matches rank above substring, mirroring the web composer's ranking); arrows or `Ctrl-n`/`Ctrl-p` move the highlight; `Enter` or `Tab` inserts the file, `Esc` closes the picker and keeps it closed while you keep typing in that same token.

Selected files are inserted as `:file[<path>]`, the exact serialization assistant-ui's default directive formatter produces on the web and sends verbatim to the daemon, so both surfaces hand the agent identical prompt text.

Design notes:
- Trigger detection, fuzzy ranking, and token replacement are pure char-index helpers in a new `mention.rs`, so multi-byte paths and mid-token cursors replace the whole `@…` run correctly.
- The `cockpit/files` response struct moved into `protocol.rs` as a shared `Serialize + Deserialize` type (next to `ReplayResponse`) so the TUI client deserializes the same shape the server emits.
- Key routing stays in the pure `dispatch` via a new `ComposerOverlay` enum rather than a bare bool, leaving room for the planned `/`-command picker (issue #1113 follow-up) without growing the argument list into boolean soup.

Closes #1699

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] In a cockpit TUI session, focus the composer and type `@`; confirm a picker appears listing project files.
- [ ] Keep typing part of a path; confirm the list narrows and prefix matches sort first.
- [ ] Press `↓`/`↑` (or `Ctrl-n`/`Ctrl-p`) to move the highlight, then `Enter` (or `Tab`); confirm the path is inserted as `:file[<path>]` at the cursor and the picker closes.
- [ ] Open the picker, press `Esc`; confirm it closes without inserting and the composer keeps focus, and that typing more in the same token does not reopen it until a fresh `@`.
- [ ] In a workspace with over 5000 files, confirm the picker shows the "list capped" footer note.

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / cockpit (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow (web composer already had the picker; only the TUI and the shared server-side `FilesResponse` type changed)

**TUI / CLI (e2e test)**
- [x] Skipped a full-binary e2e test, because: there is no interactive live-cockpit-TUI e2e harness today (only `tests/e2e/serve.rs` touches the cockpit REST surface, with no session attach + fake-agent driver), and building one for this would be slow and timing-flaky. Coverage instead lands as deterministic unit/render tests at the seams: `mention.rs` (trigger detection, fuzzy ranking, multi-byte token replacement), `input.rs` dispatch (overlay key routing vs. submit/focus), a ratatui `TestBackend` render test proving the picker lists and narrows the seeded daemon file list, and `list_files` server-side unit tests (sorting, skip dirs, dotfiles, truncation cap).

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:** Investigation, plan (with a multi-model `/debate` pass on the state model, dispatch seam, test strategy, and insertion edge cases), and implementation drafted by Claude under my direction. I made the design calls, reviewed each commit, and own the testing steps above.

- [x] I am an AI Agent filling out this form (check box if true)

## Overview

Adds an `@` file-mention picker to the TUI cockpit composer. Typing `@` opens a per-session cached workspace file list fetched from the daemon, lets users fuzzy-search and navigate results with the keyboard, and inserts the selected file as `:file[<path>]`. Behavior mirrors the web composer (prefix-over-substring ranking, keyboard navigation, insertion, and dismissal with Esc).

## What changed (high level)

- New TUI mention feature: an `@` file-mention picker that opens on `@`, supports incremental fuzzy filtering, keyboard navigation (↑/↓, Ctrl-n/Ctrl-p), selection (Enter/Tab), and dismissal (Esc) with suppression while inside a dismissed token.
- Pure mention logic: new mention.rs implements trigger detection, fuzzy ranking, and token replacement robust to multi-byte paths and mid-token cursors.
- Client HTTP: added HttpClient::files to GET /api/sessions/{id}/cockpit/files and deserialize FilesResponse.
- Shared protocol: moved FilesResponse into src/cockpit/protocol.rs so server and client share the wire-format (files: Vec<String>, truncated: bool).
- TUI integration: input routing, overlay state, view state, and render logic updated to show the picker as an overlay, manage FileIndex lifecycle, and insert serialized `:file[<path>]`.
- Server: cockpit_files endpoint now uses the shared protocol type; file-walk made deterministic (sorted dirs) and respects a 5000-entry cap; tests strengthened for sorting, skipping VCS/build/dotfiles (including nested dotfiles), and truncation.
- Docs: cockpit interface docs updated to document the `@` picker and keybinds.
- Tests: unit tests for mention logic, input dispatch/overlay routing tests, ratatui render tests for picker listing and narrowing, and server-side list_files tests including exact truncation behavior.

## What moved / removed

- FilesResponse definition moved from the server handler into src/cockpit/protocol.rs (server-side local definition removed).
- New module src/tui/cockpit_view/mention.rs added; other TUI files updated/extended (input.rs, mod.rs, render.rs, state.rs).

## Benefits

- Achieves feature parity with the web composer for file mentions.
- Faster, keyboard-driven file insertion with fuzzy search and predictable ranking.
- Robust behavior with UTF-8 paths and mid-token cursor positions.
- Server-side truncation handled and tested to avoid large-list regressions.

## Technical notes (concise)

- File list fetched lazily and cached per session (FileIndex: Unloaded/Loading/Loaded/Failed).
- fuzzy_filter is case-insensitive, prefers prefix matches over substring matches, and breaks ties by shorter paths.
- Mention detection and replacement use char-index helpers to correctly handle multibyte characters and cursor positions inside tokens.
- ComposerOverlay/Intent variants route mention navigation/accept/close; Esc stores a dismissed anchor so the picker won't reopen while typing inside the same token.
- Server directory traversal sorts entries for deterministic output and enforces a 5000-entry cap; tests assert exact truncated counts and nested dotfile skipping.

### CODE REVIEW

File: src/tui/cockpit_view/render.rs [hunk 1]
```diff
@@ -13,23 +13,108 @@ use ratatui::Frame;
...
+}
+
+fn render_mention_picker(frame: &mut Frame, area: Rect, theme: &Theme, state: &CockpitViewState) {
+    let selected = state.mention.as_ref().map(|s| s.selected).unwrap_or(0);
+    let block = Block::default()
+        .borders(Borders::ALL)
+        .title(" Files (↑/↓ or Ctrl-n/p · Enter/Tab insert · Esc close) ")
+        .border_style(Style::default().fg(theme.title));
+    let inner = block.inner(area);
+    frame.render_widget(block, area);
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/tui/cockpit_view/render.rs]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/1759/commits/13787c17b321e34ceed92c27f89847ab8d44040f — rounded border + horizontal padding, matching the other TUI pickers/dialogs.

**Comment by Seluj78** [comment]
@coderabbitai both summary nitpicks addressed in https://github.com/agent-of-empires/agent-of-empires/pull/1759/commits/de3a057a41dca975a1b2e08e70157a62b6143f15: the truncation test now asserts `files.len() == 3` (exact), and `list_files_skips_vcs_build_and_dotfiles` adds a nested `sub/.env` (skipped) alongside `sub/real.rs` (kept) to pin the all-levels dotfile rule.
